hands_test.go

```diff
@@ -0,0 +1,162 @@
+package goker_test
+
+import (
+	. "github.com/sozorogami/goker"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var royalStraightFlush = NewHand(
```

Comment:
Have we looked at https://golang.org/src/strings/strings_test.go ?

There's a common pattern in test classes of the standard library:
• Declare an anonymous struct as a general test-case, inline.
• Make a bunch of test-cases (after the anonymous struct declaration), inline.
• Test all the things.

If that were to happen here, it'd be functionally the same, but it'd also be easy to move this test class from an external test package to an internal test package....

Reply:
Thanks for those gists. 👍

Interesting to hear the concept of "internal tests." In my experience, testing implementation like that was always discouraged, since it leads to brittle tests and implies your API might be hard to use. Is this style of testing more common in Go? Does some aspect of Go philosophy necessitate it?

Re: freeing up the namespace, I actually moved these hands up in scope so I could use them in other test files. I thought it'd be clearer and more convenient than having to declare a new hand of whatever type I want. I suppose that could lead to test contamination if state is shared, though. In any case, they should probably be declared in their own file.